docs/development/etcdoperatortask-design.md

```diff
@@ -0,0 +1,501 @@
+
+}
+
+```
+</details>
+
+> **Best Practice:**
+> Use `runtime.RawExtension` for the `config` field to allow each task type to define its own schema. Validate this field with an admission webhook for type safety and user feedback.
+
+---
+
+### Status Subresource
+
+The `status` field tracks the progress and outcome of each task, including state transitions, errors, and operation history.
+
+<details>
+<summary>Show EtcdOperatorTaskStatus Go Definition</summary>
+
+```go
+// EtcdOperatorTaskStatus is the status for a EtcdOperatorTask resource.
+type EtcdOperatorTaskStatus struct {
+  // State is the last known state of the task.
+  State TaskState `json:"state"`
+  // Time at which the task has moved from "pending" state to any other state.
+  // +optional
+  InitiatedAt *metav1.Time `json:"initiatedAt"`
+  // LastError represents the errors when processing the task. Will have a limit of 10 entries at a time.
+  // +optional
+  LastErrors []LastError `json:"lastErrors,omitempty"`
+  // Captures the last operation status if task involves many stages.
+  // +optional
+  LastOperation *LastOperation `json:"lastOperation,omitempty"`
+}
+
+type LastOperation struct {
+  // Status of the last operation, one of pending, progress, completed, failed.
+  State OperationState `json:"state"`
+  // LastTransitionTime is the time at which the operation state last transitioned from one state to another.
+  LastTransitionTime *metav1.Time `json:"lastTransitionTime"`
+  // A human readable message indicating details about the last operation.
+  Description string `json:"description"`
+}
+
+// LastError stores details of the most recent error encountered for the task.
+type LastError struct {
+  // Code is an error code that uniquely identifies an error.
+  Code ErrorCode `json:"code"`
+  // Description is a human-readable message indicating details of the error.
+  Description string `json:"description"`
+  // ObservedAt is the time at which the error was observed.
+  ObservedAt *metav1.Time `json:"observedAt"`
+}
+
+// TaskState represents the state of the task ie the status of the CR.
+type TaskState string
+
+const (
+  TaskStateFailed TaskState = "Failed"
+  TaskStatePending TaskState = "Pending"
+  TaskStateRejected TaskState = "Rejected"
+  TaskStateSucceeded TaskState = "Succeeded"
+  TaskStateInProgress TaskState = "InProgress"
+)
+
+// OperationState represents the state of last operation run via the task.
+type OperationState string
+
+const (
+  OperationStateInProgress OperationState = "InProgress"
+  OperationStateCompleted OperationState = "Completed"
+  OperationStateFailed OperationState = "Failed"
+)
+
+```
+</details>
+
+---
+
+### Example: EtcdOperatorTask YAML
+
+<details>
+<summary>Show Example EtcdOperatorTask YAML</summary>
+
+```yaml
+apiVersion: druid.gardener.cloud/v1alpha1
+kind: EtcdOperatorTask
+metadata:
+    name: <name of operator task resource>
+    namespace: <cluster namespace>
```

Comment:
```suggestion
    namespace: <namespace>
```

Is it necessary that the `EtcdOperatorTask` be run in the same namespace as the referenced `Etcd` resource?